app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt

```diff
@@ -34,13 +34,67 @@ class NewsFragment @Inject constructor() : WolmoFragment<FragmentNewsBinding, Ne
             swipeRefresh.setOnRefreshListener(OnRefreshListener {
                 presenter.onRefreshSwipe()
             })
+            recyclerView.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+                val visibleThreshold = 15
+                var firstVisibleItem = 0
+                var visibleItemCount = 0
+                var totalItemCount = 0
+
+                override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+                    super.onScrolled(recyclerView, dx, dy)
+
+                    visibleItemCount = recyclerView.childCount
+                    totalItemCount = layoutManager.itemCount
+                    firstVisibleItem = layoutManager.findFirstVisibleItemPosition()
+
+                    if (totalItemCount - visibleItemCount <= firstVisibleItem + visibleThreshold) {
+                        presenter.onScrolledToEnd(currentPage)
+                    }
+                }
+            })
         }
     }
 
     override fun toggleLoadingOff() {
         binding.swipeRefresh.isRefreshing = false
     }
 
+    override fun showNews(response: NewsPage?, userId: Int) {
+        with(binding) {
+            currentPage = response?.currentPage!!
+            newsAdapter = response?.page?.let { NewsAdapter(it, userId) }
+            recyclerView.adapter = newsAdapter
+            if (recyclerView.adapter != null) {
+                if (recyclerView.adapter!!.itemCount > 0) {
+                    recyclerView.visibility = View.VISIBLE
+                    noNewsMessage.visibility = View.GONE
```

Comment:
You can add a View extension function like `View.togglePresence(visible: Boolean)` to avoid using View.VISIBLE and View.GONE every time.